### PR TITLE
XAPI plugins tests

### PR DESCRIPTION
**Pull request by benjamreis**
Signed-off-by: BenjiReis <benjamin.reis@vates.fr>

### CODE REVIEW

**Comment by benjamreis** [comment]
For now tests will only test if the plugin does not raise and some basic easy assert.
To be enhanced in a future PR.

File: tests/xapi-plugins/test_raid.py
Comment (stormi) [inline tests/xapi-plugins/test_raid.py:2]: Doesn't this raise when there's no RAID, currently?

File: tests/xapi-plugins/test_zfs.py
Comment (stormi) [inline tests/xapi-plugins/test_zfs.py:7]: I guess this will raise when there's no vol0, so the test won't work in that case?

File: tests/xapi-plugins/test_updater.py
Comment (stormi) [inline tests/xapi-plugins/test_updater.py:14]: Does yum_restore_saved_state still work when no changes have been made to the host?

File: tests/xapi-plugins/test_netdata.py
Comment (stormi) [inline tests/xapi-plugins/test_netdata.py:5]: Maybe we should assert that netdata is not installed already? Or make that a prerequisite through a fixture?

Comment (stormi) [reply tests/xapi-plugins/test_netdata.py:5]: And then we could also test what happens if you call install_netdata again when it's already installed. Two situations here by the way:
* netdata installed by install_netdata then try again
* netdata installed manually then try install_netdata (is the api key correctly set?)

File: tests/xapi-plugins/test_lsblk.py
Comment (stormi) [inline tests/xapi-plugins/test_lsblk.py:1]: The requirements should specify that you need the lsblk plugin. For convenience, you could check that the plugin exists and skip if it doesn't, as we haven't released it yet.

**Review by stormi — COMMENTED** [review]
I think each test file should have a comment that says what the requirements are, even if it's a single host with nothing in particular. BTW, since the output has changed for most of them in the master branch of xcp-ng-xapi-plugins, I suppose some of them at least require recent enough plugins. You could probably skip their execution based on the version of the xcp-ng-xapi-plugins RPM.

File: tests/xapi-plugins/test_zfs.py
Comment (benjamreis) [reply tests/xapi-plugins/test_zfs.py:7]: fixture `zpool_vol0` creates a `vol0`

Comment (stormi) [reply tests/xapi-plugins/test_zfs.py:7]: Ah, indeed

File: tests/xapi-plugins/test_netdata.py
Comment (benjamreis) [reply tests/xapi-plugins/test_netdata.py:5]: we always assume that no package are installed in other tests (mainly for storage drivers) so i think we should do the same here.
A fixture would be a good way to enforce that.

File: tests/xapi-plugins/test_updater.py
Comment (benjamreis) [reply tests/xapi-plugins/test_updater.py:14]: Good question

File: tests/xapi-plugins/test_raid.py
Comment (benjamreis) [reply tests/xapi-plugins/test_raid.py:6]: Yes, i'll add a fixture that creates the RAID if it doesn't exist and that'll remove it if it has been created for the tests.

File: tests/xapi-plugins/test_netdata.py
Comment (stormi) [reply tests/xapi-plugins/test_netdata.py]: > we always assume that no package are installed in other tests (mainly for storage drivers) so i think we should do the same here. A fixture would be a good way to enforce that.

In other tests, you don't have a plugin that may try to install things again, nor competition between users who main install it manually and the plugin that may install it, that's why I suggested that, after installing netdata once, we'd test what happens if you tell the plugin to install it again.

Comment (benjamreis) [reply tests/xapi-plugins/test_netdata.py]: I pushed a fixture enforcing netdata is NOT installed on the host. :+1:

File: tests/xapi-plugins/test_raid.py
Comment (stormi) [reply tests/xapi-plugins/test_raid.py:6]: Do I wait for this or do I approve now?

Comment (benjamreis) [reply tests/xapi-plugins/test_raid.py:6]: please wait, i'm working on it, I'd like to be sure no test raise unexpectidely before we merge. :)

Comment (benjamreis) [reply tests/xapi-plugins/test_raid.py:36]: You can approve now, after checking my fixture :)

File: tests/xapi-plugins/test_raid.py
Comment (stormi) [inline tests/xapi-plugins/test_raid.py:12]: I would prefer that you'd simply check that `/dev/md127` doesn't exist on the host.

File: tests/xapi-plugins/test_raid.py
Comment (stormi) [inline tests/xapi-plugins/test_raid.py:29]: If this fails, the files won't be deleted

File: tests/xapi-plugins/test_raid.py
Comment (stormi) [inline tests/xapi-plugins/test_raid.py:31]: Same for those

File: tests/xapi-plugins/test_raid.py
Comment (stormi) [inline tests/xapi-plugins/test_raid.py:12]: ಠ_ಠ

**Review by stormi — APPROVED** [review]

**Pull request merged**
